### PR TITLE
APM-12044 update http agent properties to improve prefix generation

### PR DESCRIPTION
**Pull request by jeff-leong**
Update http request properties to improve prefix generation
(Performance issue discovered during regression testing)
[APM-12044](https://appneta.atlassian.net/browse/APM-12044)

### CODE REVIEW

File: src/generatePrefixesList.js [hunk 1]
```diff
@@ -11,6 +11,14 @@ const clientId = "ntt-bgpalerter";
...
 import axiosEnrich from "./utils/axiosEnrich";
 
+import https from 'https';
+
+const httpsAgentOptions = {
+    keepAlive: (process.env.KEEP_ALIVE || 'true') === 'true',
+    maxSockets: parseInt(process.env.MAX_SOCKETS || '40', 10)
+}
+console.log(`Using HTTPS Agent Options: ${JSON.stringify(httpsAgentOptions)}`);
+axios.defaults.httpsAgent = new https.Agent(httpsAgentOptions);
```
Comment (jsherin) [inline src/generatePrefixesList.js:21]: @forrert made getMultipleOrigins the error it received but I cannot tell if it is necessary. Do you think you should put that back in?

Comment (jeff-leong) [reply src/generatePrefixesList.js:21]: @jsherin I don't see the change in master where Thomas made getMultipleOrigins the error it received. Can you give me a link

Comment (jsherin) [reply src/generatePrefixesList.js:21]: I am referring to: https://github.com/appneta/BGPalerter/pull/4/files#diff-e943a4a381385ea088e7616d23007ce5652c485bf88a04e25bd1710826b92df6R57

File: generatePrefixesList.js [hunk 2]
```diff
@@ -0,0 +1,427 @@
+iffdfmport axios from "axios";
```
Comment (sheniatung) [inline generatePrefixesList.js:1]: Typo?

File: generatePrefixesList.js [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+
+const apiTimeout = 120000;
+const clientId = "ntt-bgpalerter";
+const rpki = new RpkiValidator({clientId});
+import axiosEnrich from "./utils/axiosEnrich";
+
+import https from 'https';
+
+const httpsAgentOptions = {
+    keepAlive: (process.env.KEEP_ALIVE || 'true') === 'true',
```
Comment (sheniatung) [inline generatePrefixesList.js:17]: Would this always be true?

File: src/generatePrefixesList.js (hunk 1 above)
Comment (jeff-leong) [reply src/generatePrefixesList.js:21]: sounds good I added it the PR, it does not seem to introduce a problem, but I'm not sure it improves things

File: generatePrefixesList.js (hunk 3 above)
Comment (jeff-leong) [reply generatePrefixesList.js:17]: It looks like it, but I'm just propagating changes from https://github.com/appneta/BGPalerter/commit/8707d2ea2fe51f4746da62ce80665ba672a2c1bb#diff-e943a4a381385ea088e7616d23007ce5652c485bf88a04e25bd1710826b92df6R10

so I am hesitant to make the change

File: generatePrefixesList.js (hunk 2 above)
Comment (jeff-leong) [reply generatePrefixesList.js]: fixed

File: generatePrefixesList.js [hunk 4]
```diff
@@ -0,0 +1,427 @@
+import axios from "axios";
```
Comment (MagnusHvidsten) [inline generatePrefixesList.js:1]: I'm a bit confused.. Why do we want two generatePrefixesList.js files?

Comment (jeff-leong) [reply generatePrefixesList.js]: You're right, this file was added by mistake and should be removed, it is not used

Comment (MagnusHvidsten) [reply generatePrefixesList.js]: But you were making changes in /generatePrefixesList.js that were not in /src/generatePrefixesList.js... so now you've lost those changes by deleting the file?

Comment (jeff-leong) [reply generatePrefixesList.js]: The changes in  /generatePrefixesList.js are relevant, I was confused why some of my changes where no effect. I was changing the wrong file. The /src/generatePrefixesList.js is the correct and only file being used. That is why the syntax error didn't matter before.

Comment (MagnusHvidsten) [reply generatePrefixesList.js]: ok thanks for clarifying

**Review by MagnusHvidsten — APPROVED** [review]

**Review by gilbertman — APPROVED** [review]
Looks good!

**Pull request merged**